lib/puppet/provider/mongodb_user/mongodb.rb

```diff
@@ -112,14 +115,20 @@ def exists?
 
   def password_hash=(value)
     if db_ismaster
-      cmd_json=<<-EOS.gsub(/^\s*/, '').gsub(/$\n/, '')
-      {
-          "updateUser": "#{@resource[:username]}",
-          "pwd": "#{@resource[:password_hash]}",
-          "digestPassword": false
-      }
-      EOS
-      mongo_eval("db.runCommand(#{cmd_json})", @resource[:database])
+      #if mongodb is 2.4 or less, the API call is different
+      if mongo_24?
+        #this API call requires a plaintext password and does not accept hashes like the newer versions of mongodb do
+        mongo_eval("db.changeUserPassword('#{@resource[:username]}', '#{@resource[:password_plain]}')", @resource[:database])
```

Comment:
the call to change the password for mongo<=2.4 is different; it is db.changeUserPassword, and uses the plaintext password